src/sparse_blas/backends/mkl_common/mkl_operations.cxx

```diff
@@ -47,10 +47,15 @@ sycl::event optimize_gemv(sycl::queue& queue, transpose transpose_val,
                                               dependencies);
 }
 
-sycl::event optimize_trsv(sycl::queue& /*queue*/, uplo /*uplo_val*/, transpose /*transpose_val*/,
-                          diag /*diag_val*/, detail::matrix_handle* /*handle*/,
-                          const std::vector<sycl::event>& /*dependencies*/) {
-    throw unimplemented("SPARSE_BLAS", "optimize_trsv");
+sycl::event optimize_trsv(sycl::queue& queue, uplo uplo_val, transpose transpose_val, diag diag_val,
+                          detail::matrix_handle* handle,
+                          const std::vector<sycl::event>& dependencies) {
+    if (transpose_val != transpose::nontrans) {
+        throw mkl::unimplemented("sparse_blas/backends/mkl", __FUNCTION__,
+                                 "Transposed or conjugate trsv is not supported");
+    }
```

Comment:
Add `TODO` comment and replace `trsv` in thrown text with `optimize_trsv`:
```suggestion
    // TODO: Remove this if condition once Intel oneMKL adds support for
    // trans/conjtrans to optimize_trsv
    if (transpose_val != transpose::nontrans) {
        throw mkl::unimplemented("sparse_blas/backends/mkl", __FUNCTION__,
                                 "Transposed or conjugate optimize_trsv is not supported");
    }
```

Reply:
Done in https://github.com/oneapi-src/oneMKL/pull/407/commits/e49f285b7e41f354c6423359238ce2ace2d9b50d